### PR TITLE
new comparison example

### PR DESCRIPTION
**Pull request by 0ctagon**

### CODE REVIEW

File: docs/basics/1d_hist.rst [hunk 1]
```diff
@@ -218,3 +218,99 @@ To compare two histograms, using the histograms defined above:
    :alt: Simple hist
    :width: 500
 
+
+Comparing two histograms (advanced)
+===================================
+
+In this example, we will compare two tuple of histograms and use pull and diff comparisons.
```
Comment (cyrraz) [inline docs/basics/1d_hist.rst:225]: This is not a difference, but a ratio

File: docs/basics/1d_hist.rst [hunk 2]
```diff
@@ -218,3 +218,99 @@ To compare two histograms, using the histograms defined above:
...
+
+    # Harmonize the range of each axes
+    ax_main.set_xlim(x_range)
+    ax1_comparison.set_xlim(x_range)
+    ax2_comparison.set_xlim(x_range)
+
+    # Set the labels for the different axes
+    ax_main.set_ylabel("Candidates")
+    ax1_comparison.set_ylabel("$Pull_{A}$")
+    ax2_comparison.set_ylabel("$Diff_{B}$")
```
Comment (cyrraz) [inline docs/basics/1d_hist.rst:298]: This is not a difference, but a ratio

File: docs/basics/1d_hist.rst [hunk 3]
```diff
@@ -218,3 +218,99 @@ To compare two histograms, using the histograms defined above:
...
+
+    # Assign each axes: 1 to plot the histograms and 2 for the comparison plots
+    ax_main, ax1_comparison, ax2_comparison = axes
+
+    # Here, we use step as a histtype to only draw the line
+    plot_hist(h1, label="Train A", ax=ax_main, histtype="step", linewidth=1.2)
+    plot_hist(h3, label="Train B", ax=ax_main, histtype="step", linewidth=1.2)
+    # And then, to make the plot easier to read, we redraw them with stepfilled, which add color below the line
+    plot_hist(h1, ax=ax_main, histtype="stepfilled", color="#348ABD", alpha=0.2)
+    plot_hist(h3, ax=ax_main, histtype="stepfilled", color="#E24A33", alpha=0.2)
```
Comment (cyrraz) [inline docs/basics/1d_hist.rst:279]: Can we save lines with
https://stackoverflow.com/questions/28398200/plotting-transparent-histogram-with-non-transparent-edge

File: docs/basics/1d_hist.rst [hunk 4]
```diff
@@ -218,3 +218,99 @@ To compare two histograms, using the histograms defined above:
...
+    # Second comparison is using the default "difference". Same strategy as pulls
+    plot_comparison(h2, h1, ax=ax2_comparison, color="#E24A33", alpha=0.7)
+
+    # Harmonize the range of each axes
+    ax_main.set_xlim(x_range)
+    ax1_comparison.set_xlim(x_range)
+    ax2_comparison.set_xlim(x_range)
+
+    # Set the labels for the different axes
+    ax_main.set_ylabel("Candidates")
```
Comment (cyrraz) [inline docs/basics/1d_hist.rst:296]: `"Candidates"` sounds too political, let's be more moderate in our opinions and use `"Entries"`

File: docs/basics/1d_hist.rst [hunk 5]
```diff
@@ -218,3 +218,99 @@ To compare two histograms, using the histograms defined above:
...
+
+    name = "variable_1"
+    category = "category"
+
+    x1 = df[name][df[category] == 1]
+    x2 = df[name][df[category] == 4]
+    x3 = df[name][df[category] == 3]
+    x4 = df[name][df[category] == 5]
+
+    x_range = [-9, 9]
```
Comment (cyrraz) [inline docs/basics/1d_hist.rst:247]: Why not `(-10, 10)`?

Comment (cyrraz) [reply docs/basics/1d_hist.rst:247]: (to have a nicer x axis)

**Review by cyrraz — COMMENTED** [review]
Beautiful plot

File: docs/basics/1d_hist.rst [hunk 6]
```diff
@@ -218,3 +218,99 @@ To compare two histograms, using the histograms defined above:
    :alt: Simple hist
    :width: 500
 
+
+Comparing two histograms (advanced)
```
Comment (cyrraz) [inline docs/basics/1d_hist.rst:222]: Should not be in Basic 1D examples, but in a new section called Advanced 1D examples

File: docs/basics/1d_hist.rst (hunk 3 above)
Comment (0ctagon) [reply docs/basics/1d_hist.rst:279]: I'm not a fan on this complicated RGB no-one ever used `fc` argument vs this easy to read easy to understand easy to replicate example.

Comment (cyrraz) [reply docs/basics/1d_hist.rst]: Simpler: `plot_hist(h1, ax=ax_main,  histtype="stepfilled", color="#348ABD30", edgecolor="#348ABD", linewidth=1.2)`

**Review by cyrraz — APPROVED** [review]

**Pull request merged**